target/linux/ath79/image/generic.mk

```diff
@@ -355,6 +355,29 @@ define Device/dlink_dir-859-a1
 endef
 TARGET_DEVICES += dlink_dir-859-a1
 
+define Device/dlink_dir-842-c1
+  ATH_SOC := qca9563
+  DEVICE_VENDOR := D-Link
+  DEVICE_MODEL := DIR-842
+  DEVICE_VARIANT := C1
+  KERNEL := kernel-bin | append-dtb | relocate-kernel | lzma
```

Comment:
We could move everything downwards from here (and `ATH_SOC`) with the exception of the `DEVICE_PACKAGES` into a common `dlink_dir-842-c` definition, as there's quite a lot of redundancy.

Reply:
Good idea. One should think about whether this should be "dlink_dir-842-c" or "dlink_dir-842" and then also name the DTSI accordingly ...
And DEVICE_VENDOR can also be moved.

Reply:
I think dlink_dir-842-c is better?

Reply:
I would go for `dlink_dir-842-c`, as D-Link seems to base their revisions with the same letter on the same board.

Reply:
Ok, new changes committed
Thanks!

Reply:
Actually - i was still used to `DEVICE_TITLE` - you can also  move `DEVICE_VENDOR` and `DEVICE_MODEL` to the shared definition. Sorry for this additional step.

Reply:
Changed, Thanks!